File: ClassProject/Ex076IntroActivity/app/src/main/java/com/inah_wook/ex076introactivity/IntroActivity.java
package com.inah_wook.ex076introactivity;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;
import android.widget.TextView;

public class IntroActivity extends AppCompatActivity {

    // layout.xml 만들고 anim.xml 만들었으면
    //여기서 이제 이어주기 작업 고고
    // 아 맞다 ! 액티비티 화면 순서는 매니페스트가서 작업 하는건데ㅡ
    // intent-filter 있는 애가 맨 첨 뜸 ~
    //암튼 하고 왔으면 작업 시작하쟝 ~

ImageView iv;
TextView tv;

    Animation ani, ani2;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_intro);
        //에니메이션 객체 생성 : xml 에 생성한고

        ani= AnimationUtils.loadAnimation(this,R.anim.apear_logo);
        iv = findViewById(R.id.iv);
        iv.startAnimation(ani);

        // 에니메이션에 대한 동작에 대한걸 확인하는 에니메이션 리스너
        ani.setAnimationListener(new Animation.AnimationListener() {

            // 에니메이션 시작 시 하고픈 일 기입
            @Override
            public void onAnimationStart(Animation animation) {

            }

            // 에니메이션 끝났을때 하고픈 일 기입
            @Override
            public void onAnimationEnd(Animation animation) {

                // 다음으로 넘어가는 버튼을 만들거나~
                // 아예 다른 화면으로 넘어가도록 ~
                Intent intent = new Intent(IntroActivity.this, MainActivity.class);
                startActivity(intent);
                finish();
            }

            //애니메이션이 반독될 때 호출
            @Override
            public void onAnimationRepeat(Animation animation) {

            }
        });



tv=findViewById(R.id.tv);

ani2 = AnimationUtils.loadAnimation(this,R.anim.apear_title);
tv.startAnimation(ani2);

    }
}

File: ClassProject/Ex096KotlinViewTestInah/app/src/main/java/com/inah_wook/ex096kotlinviewtestinah/IntroActivity.kt
package com.inah_wook.ex096kotlinviewtestinah

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button

class IntroActivity : AppCompatActivity() {

    //맴버변수  ( 코틀린은 반드시 맴버변수 초기화 해야 함. )
    var btn01:Button? = null   // nullable 변수

    //   -> 늦은 초기화로 나중에 초기화 ~
    lateinit var btn02:Button


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_intro)

        btn01 = findViewById(R.id.btn01)
        // 익명클래스로 리스너 처리해보자
        // 코틀린은 new라는 키워드 자체가 없음
        // 또 ! nullable 변수를 사용할 땐  NullSafe 연산자를 꼭 써야 함 !!!    ->    [    ?.    ] : null이면 아예 실행을 안함.
        btn01?.setOnClickListener ( object:View.OnClickListener{
            override fun onClick(p0: View?) {
                // 자바에서 메인 실행할 때
                //  Intent intent = new Intent(IntroActivity.this, MainActivity.class)

                // 코틀린에선 ~
                val intent: Intent = Intent( this@IntroActivity, MainActivity::class.java)
                startActivity(intent)
            }

        })    // btn01 리스너

        // 근데 이렇게 하니까 초기화 졸라 귀찮음
        //   -> 늦은 초기화

        btn02 = findViewById(R.id.btn02)
        // 익명 클래스보다 표기법이 간소한 SAM 변환 사용해보쟈~ 람다식이랑 비슷~~
        // 메소드가 한개일때,
        btn02.setOnClickListener{
            finish()
        }







    }
}